packages/compartment-mapper/src/archive.js

```diff
@@ -293,19 +297,24 @@ const digestLocation = async (powers, moduleLocation, options) => {
   /** @type {Sources} */
   const sources = Object.create(null);
 
+  const internalExitModuleImportHook = exitModuleImportHookMaker({
+    modules: exitModules,
+    exitModuleImportHook,
+  });
+
   const makeImportHook = makeImportHookMaker(
     read,
     packageLocation,
     sources,
     compartments,
-    exitModules,
+    internalExitModuleImportHook,
+    true,
```

Comment:
I was very much tempted to refactor makeImportHookMaker to accept an object (I changed arguments list and structure 5 times before landing here) and I see I needlessly resisted. 
I'll happily change it.

Reply:
updated with a commit that fixes most issues bar one - with archive import hook maker. 
Today in the Endo meeting we decided to remove the ArchiveImportHookMaker type as it can't successfully be used to guard against anything.

Reply:
That's also been updated now. The refactoring is done.